app/src/main/java/org/cru/godtools/activity/BasePlatformActivity.java

```diff
@@ -242,15 +244,36 @@ private void setupNavigationDrawer() {
         }
     }
 
+    /**
+     * This method is used to update the Navigation Draws Login settings.
+     * It will only show log in information if the device is set to english
+     * and display log in links based on users status.
+     * Updated by:
+     * @author Gyasi Story
+     */
     private void updateNavigationDrawerMenu() {
-        if (mLoginItem != null) {
-            mLoginItem.setVisible(mTheKey.getDefaultSessionGuid() == null);
-        }
-        if (mSignupItem != null) {
-            mSignupItem.setVisible(mTheKey.getDefaultSessionGuid() == null);
-        }
-        if (mLogoutItem != null) {
-            mLogoutItem.setVisible(mTheKey.getDefaultSessionGuid() != null);
+        if (!ConfigurationCompat.getLocales(Resources.getSystem().getConfiguration())
+                .toLanguageTags().startsWith("en")) {  // For non English
+            if (mLoginItem != null) {
+                mLoginItem.setVisible(false);
+            }
+            if (mSignupItem != null) {
+                mSignupItem.setVisible(false);
+            }
+            if (mLogoutItem != null) {
+                mLogoutItem.setVisible(false);
+            }
+
+        } else { // For English
+            if (mLoginItem != null) {
+                mLoginItem.setVisible(mTheKey.getDefaultSessionGuid() == null);
```

Comment:
Style note: I would refactor out a local boolean called something like `isLoggedIn` and set it to the result of `mTheKey.getDefaultSessionGuid() == null`. That prevents the same logic's being evaluated three times :)

Reply:
Good point.